extensions/gitpod-web/src/extension.ts

```diff
@@ -399,87 +356,191 @@ export class GitpodWorkspaceTreeDataProvider implements vscode.TreeDataProvider<
+            </html>`;
+	}
+
+	private tunnelsMap = new Map<number, vscode.TunnelDescription>();
+	updateTunnels(tunnelsMap: Map<number, vscode.TunnelDescription>): void {
+		this.tunnelsMap = tunnelsMap;
+		this.update();
+	}
+
+	private portStatus: PortsStatusResponse | undefined;
+	updatePortsStatus(portsStatus: PortsStatusResponse): void {
+		this.portStatus = portsStatus;
+		this.update();
+	}
+
+	private updating = false;
+	private update(): void {
+		if (this.updating) { return; }
+		this.updating = true;
+		try {
+			if (!this.portStatus) { return; }
+			this.portStatus.getPortsList().forEach(e => {
+				const localPort = e.getLocalPort();
+				const gitpodPort = this.portMap.get(localPort);
+				const tunnel = this.tunnelsMap.get(localPort);
+				if (!gitpodPort) {
+					this.portMap.set(localPort, new GitpodWorkspacePort(localPort, this.context, e, tunnel));
+					return;
+				}
+				const prevStatus = gitpodPort.status;
+				gitpodPort.update(e, tunnel);
+				if (isExposedServedGitpodWorkspacePort(gitpodPort) && !isExposedServedPort(prevStatus)) {
+					this.onDidExposeServedPortEmitter.fire(gitpodPort);
+				}
+			});
+			this.onDidChangePortsEmitter.fire(this.portMap);
+			this.updateHtml();
+		} finally {
+			this.updating = false;
+		}
+	}
+
+	private updateHtml(): void {
+		const ports = Array.from(this.portMap.values()).map(e => e.toSvelteObject());
+		this._view?.webview.postMessage({ command: 'updatePorts', ports });
+	}
+
+	private onHtmlCommand() {
+		this._view?.webview.onDidReceiveMessage(async (message: { command: PortCommand; port: { info: PortInfo; status: PortsStatus.AsObject } }) => {
+			if (message.command === 'queryPortData') {
+				this.updateHtml();
+				return;
+			}
+			const port = this.portMap.get(message.port.status.localPort);
+			if (!port) { return; }
+			if (message.command === 'urlCopy' && port.status.exposed) {
+				await vscode.env.clipboard.writeText(port.status.exposed.url);
+				return;
+			}
+			vscode.commands.executeCommand('gitpod.ports.' + message.command, { port, isWebview: true });
+		});
+	}
+}
+
+export function getNonce() {
+	let text = '';
+	const possible =
+		'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
+	for (let i = 0; i < 32; i++) {
+		text += possible.charAt(Math.floor(Math.random() * possible.length));
+	}
+	return text;
+}
+
+interface PortItem { port: GitpodWorkspacePort; isWebview?: boolean }
+
 export function registerPorts(context: GitpodExtensionContext): void {
-	const gitpodWorkspaceTreeDataProvider = new GitpodWorkspaceTreeDataProvider(context);
-	const workspaceView = vscode.window.createTreeView('gitpod.workspace', {
-		treeDataProvider: gitpodWorkspaceTreeDataProvider,
-	});
-	context.subscriptions.push(workspaceView);
+
+	const experimentCfg = vscode.workspace.getConfiguration('gitpod.experiment');
+	const isExperimentMode = vscode.version.includes('insider') || experimentCfg.get<boolean>('enable') === true;
```

Comment:
Don't check for insider, only user config flag `gitpod.experimental.portsWebview.enabled`

Reply:
Speak with Anton that insiders will have this port view, I add this config to make stable can enable it too

Reply:
I was thinking by default should `true` in `gp-code/main`, but when we create the release branch we set it to `false`, no need for additional `insider` check
cc @akosyakov

Reply:
If so, we need to add this step (turn default `true` into `false) into release steps

Reply:
> I was thinking by default should true in gp-code/main, but when we create the release branch we set it to false, no need for additional insider check

I'll go with this if we can make changes in insider release. cc @jeanp413